clients/admin-ui/src/features/system/history/helpers.tsx

```diff
@@ -245,3 +266,96 @@ export const alignPrivacyDeclarations = (
     },
   };
 };
+
+/** Makes sure the before and after custom_field objects have the same keys
+ * to align the rendering of the fields in the diff modal */
+export const alignSystemCustomFields = (
+  history: SystemHistoryResponse
+): SystemHistoryResponse => {
+  const beforeCustomFields = { ...history.before.custom_fields };
+  const afterCustomFields = { ...history.after.custom_fields };
+
+  const allKeys = new Set([
+    ...Object.keys(beforeCustomFields),
+    ...Object.keys(afterCustomFields),
+  ]);
+
+  allKeys.forEach((key) => {
+    if (!(key in beforeCustomFields)) {
+      beforeCustomFields[key] = null;
+    }
+    if (!(key in afterCustomFields)) {
+      afterCustomFields[key] = null;
+    }
+  });
+
+  return {
+    ...history,
+    before: {
+      ...history.before,
+      custom_fields: beforeCustomFields,
+    },
+    after: {
+      ...history.after,
+      custom_fields: afterCustomFields,
+    },
+  };
+};
+
+/** Makes sure the custom fields object on privacy declarations have the same keys */
+export const alignPrivacyDeclarationCustomFields = (
+  history: SystemHistoryResponse
+): SystemHistoryResponse => {
+  // If privacy_declarations[0].custom_fields isn't defined, return the unmodified history object
+  if (
+    !history.before.privacy_declarations ||
+    !history.before.privacy_declarations[0] ||
+    !history.before.privacy_declarations[0].custom_fields
+  ) {
+    return history;
+  }
+
+  const beforeCustomFields = {
+    ...history.before.privacy_declarations[0].custom_fields,
+  };
+  const afterCustomFields =
+    history.after.privacy_declarations && history.after.privacy_declarations[0]
+      ? { ...history.after.privacy_declarations[0].custom_fields }
+      : {};
+
+  const allKeys = new Set([
+    ...Object.keys(beforeCustomFields),
+    ...Object.keys(afterCustomFields),
+  ]);
+
+  allKeys.forEach((key) => {
```

Comment:
I think we may have the same sort order problem here, though I didn't encounter it specifically